actions/user_actions.jsx

```diff
@@ -39,45 +39,58 @@ export async function switchFromLdapToEmail(email, password, token, ldapPassword
+    return async (doDispatch, doGetState) => {
+        const newTeamId = teamId || getCurrentTeamId(doGetState());
+        const {data} = await doDispatch(UserActions.getProfilesInTeam(newTeamId, page, perPage));
+        if (data) {
+            doDispatch(loadTeamMembersForProfilesList(data, newTeamId));
+            doDispatch(loadStatusesForProfilesList(data));
+        }
 
-export async function loadProfilesAndTeamMembersAndChannelMembers(page, perPage, teamId = getCurrentTeamId(getState()), channelId = getCurrentChannelId(getState()), success, error) {
-    const {data} = await UserActions.getProfilesInChannel(channelId, page, perPage)(dispatch, getState);
+        return {data: true};
+    };
+}
 
-    loadTeamMembersForProfilesList(
-        data,
-        teamId,
-        () => {
-            loadChannelMembersForProfilesList(data, channelId, success, error);
-            dispatch(loadStatusesForProfilesList(data));
+export function loadProfilesAndTeamMembersAndChannelMembers(page, perPage, teamId, channelId) {
+    return async (doDispatch, doGetState) => {
+        const state = doGetState();
+        const teamIdParam = teamId || getCurrentTeamId(state);
+        const channelIdParam = channelId || getCurrentChannelId(state);
+        const {data} = await doDispatch(UserActions.getProfilesInChannel(channelIdParam, page, perPage));
+        if (data) {
+            doDispatch(loadTeamMembersForProfilesList(data, teamIdParam)).then(() => {
+                doDispatch(loadChannelMembersForProfilesList(data, channelIdParam));
+                doDispatch(loadStatusesForProfilesList(data));
+            });
         }
-    );
-}
 
-export function loadTeamMembersForProfilesList(profiles, teamId = getCurrentTeamId(getState()), success, error) {
-    const state = getState();
-    const membersToLoad = {};
-    for (let i = 0; i < profiles.length; i++) {
-        const pid = profiles[i].id;
+        return {data: true};
+    };
+}
 
-        if (!getTeamMember(state, teamId, pid)) {
-            membersToLoad[pid] = true;
+export function loadTeamMembersForProfilesList(profiles, teamId) {
+    return async (doDispatch, doGetState) => {
+        const state = doGetState();
+        const teamIdParam = teamId || getCurrentTeamId(state);
+        const membersToLoad = {};
+        for (let i = 0; i < profiles.length; i++) {
+            const pid = profiles[i].id;
+
+            if (!getTeamMember(state, teamIdParam, pid)) {
+                membersToLoad[pid] = true;
+            }
         }
-    }
 
-    const list = Object.keys(membersToLoad);
-    if (list.length === 0) {
-        if (success) {
-            success({});
+        const userIdsToLoad = Object.keys(membersToLoad);
+        if (userIdsToLoad.length === 0) {
+            return {data: true};
         }
-        return;
-    }
 
-    loadTeamMembersForProfiles(list, teamId, success, error);
+        doDispatch(getTeamMembersByIds(teamIdParam, userIdsToLoad));
```

Comment:
Same here, the `.then` is called before the `getTeamMembersByIds` returns, with the other implementation is called after it returns. It can be solved adding an `await` here.

Reply:
You're right.  I'll add it.